### PR TITLE
[MAD][DATA 2018][Lab-intro-ml] Constan M. V. 

### PR DESCRIPTION
**Pull request by cmvalma**
El ejercicio se ve más elaborado que otros anteriores pero en vez de hacer un ejercicio de SVM, que no lo hemos visto, no estaría mal hacerlo sobre lo que sí hemos visto...
Fuera de esto,  he usado gamma = scalar por indicación de scikit y funciona, pero me gustaría saber qué es... 
Salut.

### CODE REVIEW

**Comment by Gobinde43** [comment]
> El ejercicio se ve más elaborado que otros anteriores pero en vez de hacer un ejercicio de SVM, que no lo hemos visto, no estaría mal hacerlo sobre lo que sí hemos visto...
> Fuera de esto, he usado gamma = scalar por indicación de scikit y funciona, pero me gustaría saber qué es...
> Salut.

Gracias Constan por el feedback!! Transmito tus comentarios y te invito a que los incorpores en tu survey de la semana.

File: module-3/lab-intro-to-ml/your-code/main.ipynb [hunk 1]
```diff
@@ -12,11 +12,16 @@
   },
```
Comment (Gobinde43) [inline module-3/lab-intro-to-ml/your-code/main.ipynb]: - Good one Constan!! 
```Python
austin[wrong_type_columns] = austin[wrong_type_columns].apply(pd.to_numeric, errors = 'coerce')
```
- "Next, count the number of rows of austin and missing_values." Try to use size method and check the differences with len one. 
```Python
print(austin.size)
print(missing_values.size)
```
- Why are you using index in the following? 
```Python
len(austin.index)/len(missing_values.index)
# you only need to do something like the following, using size or len
missing_values.size / austin.size
```
- Your code for this "As you can see from the output, the majority of missing data is in one column called PrecipitationSumInches. What's the number of missing values in this column in ratio to its total number of rows?" is not correct. 
```Python 
#Instead of yours 
austin.PrecipitationSumInches.isna().sum()/ len(austin.PrecipitationSumInches.index)
#just try 
missing_values['PrecipitationSumInches'].size / austin['PrecipitationSumInches'].size
```
- This "In the cell below, convert the Date column values from string to numeric values using toordinal()" could be as simple as the following:
```Python
from datetime import datetime as dt
austin_fixed['Date'] = austin_fixed['Date'].apply(lambda date_col: dt.strptime(date_col, '%Y-%m-%d').toordinal())
```
- You can do this "Your y should be a subset of austin_fixed containing one column TempAvgF" in a more simple way. It could be something similar to the following:
```Python
response_col_names = list(austin_fixed.columns.values)
response_col_names.remove('TempAvgF')
X = austin_fixed[response_col_names]
y = austin_fixed['TempAvgF']
```
- Gamma and C parameters for SVM (rbf and poly, not linear). The default in SVC is rbf. Try to train both a linear a
https://www.youtube.com/watch?v=m2a2K4lprQw
https://www.youtube.com/watch?v=joTa_FeMZ2s
- If you want to play with synthetic data for classification problems, scikit-learn has a function to generate synthetic datasets. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html.
It's very useful to play with different type of classifiers (two label)
```Python
from sklearn.datasets import make_classification
```
- Bonus Challenge 2: Here try to train a linear SVM instead of the rbf one you've used. Previous to training a classifier it's a good idea to plot a scatter plot of your data for each label/class. This can give you a visual idea of how well a linear or not linear classifier will perform in the training step.
